server/app/jobs/grid_scheduler_job.rb

```diff
@@ -33,12 +34,23 @@ def schedule_grids
   # @param [Grid] grid
   def schedule_grid(grid)
     scheduler = GridScheduler.new(grid)
-    grid.grid_services.order(updated_at: :asc).each do |service|
-      begin
-        scheduler.check_service(service)
-      rescue => exc
-        error "error occurred in service #{service.to_path}"
-        error exc.message
+    grid.grid_services.order(updated_at: :asc).pluck(:id).each do |service_id|
+      service = GridService.find(service_id)
+      if service && leader?
+        begin
+          ts = Time.now
+          Mongoid::QueryCache.cache {
```

Comment:
Looks like [`Mongoid::QueryCache.cache`](https://github.com/mongodb/mongoid/blob/51f063f6d35ca27d7ecabd9a1b4beceb30d2518b/lib/mongoid/query_cache.rb#L66-L72) just enables/disables the cache: it does not clear it.

Needs something like the [`Mongoid::QueryCache::Middlware#call`](https://github.com/mongodb/mongoid/blob/51f063f6d35ca27d7ecabd9a1b4beceb30d2518b/lib/mongoid/query_cache.rb#L117-L121)  to actually clear the cache in between loop iterations:

```ruby
      def with_cache(env)
        QueryCache.cache { yield }
      ensure
        QueryCache.clear_cache
      end
```

IOW that means that it's currently caching all queries for the entire lifetime of the actor thread...?

TBH I reckon that the caching could happen at the `schedule_grids` level, i.e. share the same cache across all of the services being rescheduled.

Reply:
Fixed.